docs/packaging/versioning.md

```diff
@@ -0,0 +1,140 @@
+| dev releases         | https://rocm.devreleases.amd.com/ | Manually triggered test workflows in [TheRock](https://github.com/ROCm/TheRock)                                                                                                                              |
+| dev builds           | No central index                  | Local builds and per-commit workflows in [TheRock](https://github.com/ROCm/TheRock),<br>[rocm-libraries](https://github.com/ROCm/rocm-libraries), [rocm-systems](https://github.com/ROCm/rocm-systems), etc. |
+
+## Python package versions
+
+Python package versions are handled by scripts:
+
+- [`build_tools/compute_rocm_package_version.py`](/build_tools/compute_rocm_package_version.py)
+  - [`build_tools/tests/compute_rocm_package_version_test.py`](/build_tools/tests/compute_rocm_package_version_test.py)
+
+The script produces these versions for each distribution channel:
+
+| Distribution channel | Version format    | Version example                                                                                                                                                     |
+| -------------------- | ----------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| stable releases      | `X.Y.Z`           | `7.10.0`                                                                                                                                                            |
+| prereleases          | `X.Y.ZrcN`        | `7.10.0rc0`                                                                                                                                                         |
+| nightly releases     | `X.Y.ZaYYYYMMDD`  | `7.10.0a20251124`                                                                                                                                                   |
+| dev builds/releases  | `X.Y.Z.dev0+NNNN` | `7.10.0.dev0+efed3c3b10a5cce8578f58f8eb288582c26d18c4`<br>(for commit [`efed3c3`](https://github.com/ROCm/TheRock/commit/efed3c3b10a5cce8578f58f8eb288582c26d18c4)) |
+
+Each distribution channel (and GPU family within that channel) is currently
+hosted on a separate release index that can be selected with e.g.
+`pip install --index-url=https://rocm.nightlies.amd.com/v2/gfx94X-dcgpu/ rocm`.
+See [RELEASES.md - Installing releases using pip](../../RELEASES.md#installing-releases-using-pip)
+for details.
+
+### External Python package versions
+
+When we build external projects like
+[PyTorch](https://github.com/pytorch/pytorch) we sometimes extend the base
+package version with our own
+[local version identifier](https://packaging.python.org/en/latest/specifications/version-specifiers/#local-version-identifiers).
+
+For example, for torch version `2.9.0` built with ROCm version `7.9.0` we
+generate a composite torch version `2.9.0+rocm7.9.0`.
```

Comment:
These local versions may not work in all cases we hope/expect them to, see https://github.com/ROCm/TheRock/issues/2368

Reply:
Added a TODO to circle back to that and update the doc.